src/data-access/FetchParseValueRepository.ts

```diff
@@ -0,0 +1,70 @@
+import ParseValueRepository from '@/data-access/ParseValueRepository';
+import ParseResult from '@/data-access/ParseResult';
+import TechnicalProblem from '@/data-access/errors/TechnicalProblem';
+import PrecisionError from '@/data-access/errors/PrecisionError';
+
+export default class FetchParseValueRepository implements ParseValueRepository {
+	private readonly endpoint: string;
+
+	public constructor( endpoint: string ) {
+		this.endpoint = endpoint;
+	}
+
+	public async parseValues( values: string[], datatype: string ):
+	Promise<ParseResult[]> {
+		if ( !values ) {
+			throw new Error( 'The parameter values must not be empty!' );
+		}
+
+		// NOTE: remove this line if/when we decide to supported with multiple values.
+		if ( values.length > 1 ) {
+			throw new Error( 'only one value is supported.' );
+		}
+
+		if ( !datatype ) {
+			throw new Error( 'The parameter datatype must not be empty!' );
+		}
+
+		const concatinateValues = values.reduce( ( accumulator, currentValue ) => accumulator + '|' + currentValue );
```

Comment:
You can use `.join( '|' )` instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join

Reply:
ah, thanks. thats much cleaner